sdk/encrypt.go

```diff
@@ -0,0 +1,87 @@
+package sdk
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"io"
+)
+
+func Encrypt(keyReq KeyRequest, input []byte) (string, error) {
+	capeKey, err := Key(keyReq)
+	if err != nil {
+		return "", err
+	}
+
+	dataCiphertext, key, err := AESEncrypt(input)
+	if err != nil {
+		return "", err
+	}
+
+	keyCiphertext, err := RSAEncrypt(key, capeKey)
+	if err != nil {
+		return "", err
+	}
+
+	ciphertext := base64.StdEncoding.EncodeToString(append(keyCiphertext, dataCiphertext...))
+
+	capeEncryptPrefix := "cape:"
+	result := capeEncryptPrefix + ciphertext
+
+	return result, nil
+}
+
+func AESEncrypt(plaintext []byte) ([]byte, []byte, error) {
+	key := make([]byte, 32) // generate a random 32 byte key for AES-256
+	if _, err := rand.Read(key); err != nil {
+		return nil, nil, err
+	}
+
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	aesGCM, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	nonce := make([]byte, aesGCM.NonceSize())
+	if _, err = io.ReadFull(rand.Reader, nonce); err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: is dst needed?
+	ciphertext := aesGCM.Seal(nonce, nonce, plaintext, nil)
```

Comment:
I just left nonce as nil for the encryption module in runtime.

Reply:
By including the nonce in the first parameter, the ciphertext will embed the nonce as a prefix. This way the nonce doesn't need to be handled and stored separately. We should change runtime's deploy to do this as well

Reply:
Will we ever change the size of the nonce, since right now we add it to the end of the function struct as a separate field. I think there is a way to make this backwards compatible given what we have now. e.g.: We just try to deserialize the function struct and if the nonce is not there we extract it from the first few bytes of the payload.

Reply:
The recommended nonce size is `aesGCM.NonceSize()` (12 bytes) which is also a change that should be made on the runtime side (the value we hardcode there is 12 bytes it's just recommended to use the value that the aes lib defines)